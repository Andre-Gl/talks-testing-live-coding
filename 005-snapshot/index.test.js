// const mockData = {'TST' : 'Test'};
// const mockStatesFn = jest.fn().mockImplementation(() => mockData);
// const mockZipLookupFn = jest.fn().mockImplementation(() => mockData);
//
// jest.mock('./data', () => ({
//   getStates: mockStatesFn,
//   zipLookup:
// }));

const states = require('./index');

test('getStatesAbbrList', () => {
  expect(states.getStatesAbbrList()).toMatchSnapshot();
});

describe('zipLookup', () => {
  it('should resolve correct zip', () => {
    expect.assertions(1);
    return expect(states.zipLookup('94085')).resolves.toEqual('CA');
  });


  it('should reject on invalid zip', () => {
    expect.assertions(1);
    return expect(states.zipLookup('')).rejects.toMatch('Zip code is required');
  });
});
