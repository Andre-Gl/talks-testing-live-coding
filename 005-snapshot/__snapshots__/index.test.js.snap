// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStatesAbbrList should return list of abbreviations 1`] = `
Array [
  "AZ",
  "AL",
  "AK",
  "AR",
  "CA",
  "CO",
  "CT",
  "DC",
  "DE",
  "FL",
  "GA",
  "HI",
  "ID",
  "IL",
  "IN",
  "IA",
  "KS",
  "KY",
  "LA",
  "ME",
  "MD",
  "MA",
  "MI",
  "MN",
  "MS",
  "MO",
  "MT",
  "NE",
  "NV",
  "NH",
  "NJ",
  "NM",
  "NY",
  "NC",
  "ND",
  "OH",
  "OK",
  "OR",
  "PA",
  "RI",
  "SC",
  "SD",
  "TN",
  "TX",
  "UT",
  "VT",
  "VA",
  "WA",
  "WV",
  "WI",
  "WY",
]
`;

exports[`getStatesTypeAhead should return array of states that starts with symbols 1`] = `
Array [
  "Alabama",
  "Alaska",
]
`;

exports[`getStatesTypeAhead should return array of states that starts with symbols limitted to 3 matches 1`] = `
Array [
  "New Hampshire",
  "New Jersey",
  "New Mexico",
]
`;

exports[`getStatesTypeAhead should return empty array when no input 1`] = `Array []`;

exports[`zipLookup should reject on invalid zip 1`] = `"Zip code is required"`;

exports[`zipLookup should resolve correct zip 1`] = `"CA"`;
